File: Algorithms/java/199.BinaryTreeRightSideView.java
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        if (root == null)
            return result;
        result.add(root.val);
        List<TreeNode> cur_level = new ArrayList<>();
        List<TreeNode> next_level = new ArrayList<>();
        cur_level.add(root);
        while(cur_level.size() != 0) {
            TreeNode node = cur_level.get(0);
            cur_level.remove(0);
            if (node.left != null) next_level.add(node.left);
            if (node.right != null) next_level.add(node.right);
            if (cur_level.size() == 0) {
                cur_level = next_level;
                next_level = new ArrayList<>();
                if (cur_level.size() != 0)
                    result.add(cur_level.get(cur_level.size()-1).val);
            }
        }
        return result;
    }
}

File: Algorithms/py/199.BinaryTreeRightSideView.py
class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        result = []
        if not root:
            return result
        result.append(root.val)
        cur_level, next_level = [], []
        cur_level.append(root)
        while cur_level:
            node = cur_level.pop(0)
            if node.left:
                next_level.append(node.left)
            if node.right:
                next_level.append(node.right)
            if not cur_level:
                cur_level = next_level
                next_level = []
                if cur_level:
                    result.append(cur_level[-1].val)
        return result
